FileManager/Models/FilesManagerService.swift

```diff
@@ -0,0 +1,143 @@
+        filesData = contents.map {
+            let name = $0.lastPathComponent
+            
+            return FilesUnit(name: name,
+                      path: $0,
+                      type: name.contains(".png") || name.contains(".jpeg") ? .image : .folder)
+        }
+        
+        delegate?.reloadData()
+    }
+    
+    func addDirectory(name: String) {
+        guard let currentDirectory = currentDirectory else { return }
+        
+        let filePath = currentDirectory.appendingPathComponent(name)
+        if !fileManager.fileExists(atPath: filePath.path) {
+            do {
+                try fileManager.createDirectory(atPath: filePath.path,
+                                                withIntermediateDirectories: false,
+                                                attributes: nil)
+                updateFilesData()
+            } catch {
+                print("Error")
+            }
+        }
+    }
+    
+    func createImage(_ image: UIImage, name: String) {
+        guard let currentDirectory = self.currentDirectory,
+              let data = image.jpegData(compressionQuality: 1) else { return }
+        
+        let newImagePath = currentDirectory.appendingPathComponent(name)
+        
+        try? data.write(to: newImagePath)
+        
+        updateFilesData()
+    }
+
+    func openFile(file: FilesUnit, navigationController: UINavigationController?) {
+        let newDirectory = file.path
+        
+        switch file.type {
+        case .folder:
+            let selectFolderViewContoller = FilesViewController()
+            selectFolderViewContoller.manager.currentDirectory = newDirectory
+            navigationController?.pushViewController(selectFolderViewContoller, animated: true)
+
+        default:
+            let selectImageViewContoller = ImageViewController()
+            if let data = try? Data(contentsOf: newDirectory) {
+                selectImageViewContoller.image = UIImage(data: data)
+            }
+            navigationController?.pushViewController(selectImageViewContoller, animated: true)
+        }
+    }
+    
+    func deleteFile(file: FilesUnit) {
+        try? FileManager.default.removeItem(at: file.path)
+        
+        updateFilesData()
+    }
+    
+    func deleteFiles() {
+        guard viewMode == .select else { return }
+        
+        for file in selectedFiles {
+            try? FileManager.default.removeItem(at: file.path)
+        }
+        
+        viewMode = .view
+        
+        updateFilesData()
+    }
+    
+    func selectFile(file: FilesUnit) {
+        guard viewMode == .select else { return }
+        
+        if let index = selectedFiles.firstIndex(of: file) {
+            selectedFiles.remove(at: index)
+        }
+        else {
+            selectedFiles.append(file)
+        }
+        
+        delegate?.reloadData()
+    }
+    
+    func handleSelectionCompleted() {
+        selectedFiles.removeAll()
+        updateFilesData()
```

Comment:
Do you need to `updateFilesData()` or do you just need to `delegate?.reloadData()`